tools/dpip/ipset.c

```diff
@@ -65,28 +65,25 @@ static int ipset_parse_args(struct dpip_conf *conf, struct dp_vs_multi_ipset_con
     while (conf->argc > 0) {
         ipaddr = conf->argv[0];
         ips->ipset_conf[index].af = AF_INET;
-   	if (inet_pton_try(&conf->af, ipaddr, &ips->ipset_conf[index].addr) <= 0)
- 	{
-             fprintf(stderr, "bad IP\n");
-   	     dpvs_sockopt_msg_free(ips);
-	     return -1;
-	}
-	index++;
+        if (inet_pton_try(&conf->af, ipaddr, &ips->ipset_conf[index].addr) <= 0)
+        {
+            fprintf(stderr, "bad IP\n");
+            dpvs_sockopt_msg_free(ips);
```

Comment:
`dpvs_sockopt_msg_free` is used to free memory allocated by `dpvs_getsockopt`. It does work here. But its behavior may be changed with implementation of sockopt module.